### ISSUE TITLE
Allow confirmation of auto-approval of deleted signed add-on versions

### ISSUE THREAD

**Issue by abhn** [mozilla/addons#1948]
### Describe the problem and steps to reproduce it:
For an add-on version that was

* Signed (auto-approved)
* Deleted by the developer
* Needs to be reviewed

There's no good way to mark it as reviewed (The reviewer can add a comment, but that isn't ideal as then we need separate logic to count this add-on's review)

### What did you expect to happen?
* The "Confirm auto-approval" button works for add-on versions which were signed by AMO, even if they are deleted.
* A reviewer is able to mark an auto-approved version of an add-on as confirmed even if it is deleted

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/AMOENG-234)

**Comment by wagnerand** [comment]
@abhn is this still an issue? If so, can you provide some more context and reply to @eviljeff's questions?

**Comment by abhn** [comment]
@wagnerand haven't had this case in a while, but if nothing's changed, the next time the situation in OP comes up, we might hit this. 

@eviljeff (apologies, looks like I missed your message for a year 3:) I think it should be the "Confirm Multiple Versions" action that should include the deleted versions. 

The rationale to this is that if we can set "Needs Human Review" on a deleted version (and reviewers have to review one such version), then we should be able to follow standard processes around a signed version review even if it was deleted. Does that make sense, or am I missing out on something here?

**Comment by eviljeff** [comment]
Assuming past me was correct when he said "action is written with deleted versions in mind (there's a comment about them)." then it's a bug that deleted (unlisted) versions can't be confirmed.

**Comment by abhn** [comment]
In that case, does this issue (and the included STR) suffice to track the bug? Or would you like specific examples where this is happening (I don't have one handy or in recent memory, but if you need additional details, I can try to gather).

**Comment by eviljeff** [comment]
If it's happening with every deleted version then the STR is sufficient... if it's _not_ happening with every deleted version, but under some other circumstances, then we'll need more details and/or some links to examples to investigate.

**Comment by eviljeff** [comment]
(I'm kinda assuming it's every deleted version, but if you have evidence/recollection otherwise, that's good to know)

**Comment by abhn** [comment]
> kinda assuming it's every deleted version
Yes, I checked two more cases today (one of which is my own add-on) and it held true everywhere. "Confirm multiple versions" don't show deleted versions.

**Comment by ioanarusiczki** [comment]
@chrstinalin Is this an issue for 23.01.2025 milestone ?

**Comment by ioanarusiczki** [comment]
This is working for me on AMO dev 
https://reviewers.addons-dev.allizom.org/en-US/reviewers/review-unlisted/635500
